server/src/main/java/org/opensearch/index/translog/transfer/BlobStoreTransferService.java

```diff
@@ -41,18 +53,18 @@ public BlobStoreTransferService(BlobStore blobStore, ThreadPool threadPool) {
     }
 
     @Override
-    public void uploadBlobAsync(
-        String threadpoolName,
+    public void uploadBlobByThreadPool(
```

Comment:
we can keep the same name here.

Reply:
`uploadBlobAsync` can be confused with async IO that's why this name was changed.  I am open for other name suggestions which do not have async term in them.

Reply:
No as long the method invocation is async it doesn't really matter `byThreadPool` name is non-intuitive